integration_tests/src/main/python/string_test.py

```diff
@@ -939,3 +906,33 @@ def test_format_number_float_value():
         lambda spark: unary_op_df(spark, data_gen).selectExpr('format_number(a, 5)').collect())))
     for cpu, gpu in zip(cpu_results, gpu_results):
         assert math.isclose(cpu, gpu, rel_tol=1e-7) or math.isclose(cpu, gpu, abs_tol=1.1e-5)
+
+# valid base range is [2, 36], to_base can be negative, out of range results nulls
+# TODO: test ansi mode
+@pytest.mark.parametrize('ansi_mode', [False], ids=idfn)
+def test_conv(ansi_mode):
+    gen = [
+        ("str_col", mk_str_gen(r'-?[0-9a-zA-Z]{1,15}')),
+        ("from_col", IntegerGen(min_val=0, max_val=38)),
```

Comment:
Can you make 38 some mnemonic  constants and comment on why? I just had to remind myself about it: The alphabet for digits [a-z] + [0-9] consists of 36 symbols. Conceptually the digit interval is [0..35] , so does it not make 36 the largest representable base?  If it is to test invalid values, please add comments.

Reply:
Added comments.